### ISSUE TITLE
PATCH event results in 500

### ISSUE THREAD

**Issue by w-le** [place-labs/google-staff-api#5]
PATCH https://mysmartoffice-stg.mer.pwc.com/api/staff/v1/events/sv1afni1pjcfuov2c3oa9vi8ug?system_id=sys-FaBTfr3zFaY

which has payload
![image](https://user-images.githubusercontent.com/10395560/117127794-26757700-adcf-11eb-96e2-8e3a11097e8f.png)

note the `timezone` value, which is what Gcal responds with

500 error response is
```
ERROR: Invalid location name: -07:00 (Time::Location::InvalidLocationNameError)
  from usr/share/crystal/src/time/location.cr:57:0 in 'new'
  from src/src/controllers/events.cr:409:29 in 'update'
  from usr/share/crystal/src/int.cr:1106:11 in '???'
  from usr/share/crystal/src/primitives.cr:255:3 in 'call'
  from usr/share/crystal/src/http/server/handler.cr:28:7 in 'call'
  from usr/share/crystal/src/http/server/handler.cr:28:7 in 'call'
  from usr/share/crystal/src/http/server/request_processor.cr:51:11 in 'process'
  from usr/share/crystal/src/http/server.cr:498:5 in '->'
  from usr/share/crystal/src/primitives.cr:255:3 in 'run'
  from ???
```

**Comment by stakach** [comment]
yeah timezone needs to be an IANA timezone name:
https://en.wikipedia.org/wiki/List_of_tz_database_time_zones

not just the offset

**Comment by w-le** [comment]
Yes but annoyingly that value was what was returned by staff API when an event was created in google calander

**Comment by w-le** [comment]
here is a staff-api response to `/events` that shows a booking created from google calendar, note the timezone value:
![image](https://user-images.githubusercontent.com/10395560/117148146-8a586980-ade8-11eb-9876-aee7290bb027.png)

**Comment by w-le** [comment]
so if google calendar natively may return those types of values for `timezone` then I think our backend model would need to support it too,

otherwise we would need all frontends to manually check and convert `timezone` to IANA for every patch, even when it's not making a CHANGE to the timezone

**Comment by stakach** [comment]
that makes sense, so then that is new from the recent changes we made, I'll see if I can resolve it

**Comment by stakach** [comment]
fixed in 37b57b4
